provider/core/src/resource.rs

```diff
@@ -384,11 +383,28 @@ impl From<LanguageIdentifier> for ResourceOptions {
     }
 }
 
+impl From<Locale> for ResourceOptions {
+    /// Create a ResourceOptions with the given language identifier and an empty variant field.
+    fn from(locale: Locale) -> Self {
+        Self {
+            langid: Some(locale.id),
+            variant: None,
+        }
+    }
+}
+
 impl ResourceOptions {
```

Comment:
While we're changing the API, I think we should add constructors for `ResourceOptions` and make the fields private. Currently there is no instance that has a variant but no langid, and I think we should require that. Otherwise we cannot parse stringified `ResourceOptions`, as both

`ResourceOptions { variant: Some(Cow::Borrowed("und")), langid: None}`

`ResourceOptions { variant: None, langid: Some(langid!("und")) }`

are 'serialized' to `"und"`.

We need that for #1092.

Reply:
As discussed, I don't want to make this change right now because I am reconsidering the whole data model of ResourceOptions as part of #243.